Add build for linux-aarch64

martin-g opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

martin-g committed:
Add build for linux-aarch64

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

Commit:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.3, and conda-forge-pinning 2025.01.23.20.59.37

martin-g committed:
osx-arm64: Try by using FC_FOR_BUILD.

paraview-feedstock does this for CC_FOR_BUILD. Maybe it will help here
too

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Do not build for osx-arm64

It fails with:
```
 + make config
 │ │ + FFLAGS='-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/swan-41.51 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -fno-second-underscore -fallow-argument-mismatch'
 │ │ + FFLAGS90='-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/swan-41.51 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -fno-second-underscore -fallow-argument-mismatch -ffree-line-length-none'
 │ │ + make mpi
 │ │ make FOR=mpif90 FFLAGS="-O -w -fno-second-underscore -fallow-argument-mismatch " \
 │ │         FFLAGS90="-O -w -fno-second-underscore -fallow-argument-mismatch -ffree-line-length-none " \
 │ │                 INCS="" LIBS="" OBJS="swmod1.o swmod2.o SwanSpectPart.o m_constants.o m_fileio.o serv_xnl4v5.o mod_xnl4v5.o SwanGriddata.o SwanGridobjects.o SwanCompdata.o SdsBabanin.o SwanIEM.o SwanBraggScat.o SwanQCM.o   swan2coh.o swanmain.o swanpre1.o swanpre2.o swancom1.o swancom2.o swancom3.o swancom4.o swancom5.o swanout1.o swanout2.o swanser.o swanparll.o SwanReadGrid.o SwanReadADCGrid.o SwanReadTriangleGrid.o SwanReadEasymeshGrid.o SwanInitCompGrid.o SwanCheckGrid.o SwanCreateEdges.o SwanGridTopology.o SwanGridVert.o SwanGridCell.o SwanGridFace.o SwanPrintGridInfo.o SwanFindPoint.o SwanPointinMesh.o SwanBpntlist.o SwanPrepComp.o SwanVertlist.o SwanCompUnstruc.o SwanDispParm.o SwanPropvelX.o SwanSweepSel.o SwanPropvelS.o SwanTranspAc.o SwanTranspX.o SwanDiffPar.o SwanGSECorr.o SwanGradDepthorK.o SwanGradVel.o SwanInterpolatePoint.o SwanInterpolateAc.o SwanInterpolateOutput.o SwanConvAccur.o SwanConvStopc.o SwanThreadBounds.o SwanFindObstacles.o SwanCrossObstacle.o SwanComputeForce.o SwanIntgratSpc.o SwanBndStruc.o SwanVTKWriteHeader.o SwanVTKWriteData.o SwanVTKPDataSets.o ocpids.o ocpcre.o ocpmix.o fftpack51.o" swan.exe
 │ │ make[1]: Entering directory '$SRC_DIR'
 │ │ mpif90 swmod1.f -c -O -w -fno-second-underscore -fallow-argument-mismatch
 │ │ make[1]: mpif90: Bad CPU type in executable
 │ │ make[1]: *** [Makefile:185: swmod1.o] Error 127
 │ │ make[1]: Leaving directory '$SRC_DIR'
```

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.3, and conda-forge-pinning 2025.01.23.20.59.37
